Fix custom op bug of clearing dir

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137655
Previously when we delete a custom op out of context manager, we weren't clearing the dir field of the op namespace. As a result, it was polluting other tests. 

Differential Revision: [D64141465](https://our.internmc.facebook.com/intern/diff/D64141465/)

tugsbayasgalan committed:
Fix custom op bug of clearing dir



Differential Revision: [D64141465](https://our.internmc.facebook.com/intern/diff/D64141465/)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix custom op bug of clearing dir"

Previously when we delete a custom op out of context manager, we weren't clearing the dir field of the op namespace. As a result, it was polluting other tests. 

Differential Revision: [D64141465](https://our.internmc.facebook.com/intern/diff/D64141465/)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix custom op bug of clearing dir"

Previously when we delete a custom op out of context manager, we weren't clearing the dir field of the op namespace. As a result, it was polluting other tests. 

Differential Revision: [D64141465](https://our.internmc.facebook.com/intern/diff/D64141465/)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix custom op bug of clearing dir"

Previously when we delete a custom op out of context manager, we weren't clearing the dir field of the op namespace. As a result, it was polluting other tests. 

Differential Revision: [D64141465](https://our.internmc.facebook.com/intern/diff/D64141465/)

[ghstack-poisoned]

Skylion007 commented on test/export/test_export.py:
```diff
@@ -1047,6 +1047,8 @@ def forward(self, x):
             warnings.simplefilter("error")
             torch.export.export(Foo(), (x,))
 
+        ops_registered_before = set(op for op in torch.ops.mylib)
```
```suggestion
        ops_registered_before = set(torch.ops.mylib)
```

Skylion007 commented on test/export/test_export.py:
```diff
@@ -1063,6 +1065,9 @@ def forward(self, x):
                     warnings.simplefilter("always")
                     torch.export.export(Bar(), (x,))
 
+        ops_registered_after = set(op for op in torch.ops.mylib)
```
```suggestion
        ops_registered_after = set(torch.ops.mylib)
```

tugsbayasgalan committed:
Update on "Fix custom op bug of clearing dir"

Previously when we delete a custom op out of context manager, we weren't clearing the dir field of the op namespace. As a result, it was polluting other tests. 

Differential Revision: [D64141465](https://our.internmc.facebook.com/intern/diff/D64141465/)

[ghstack-poisoned]
